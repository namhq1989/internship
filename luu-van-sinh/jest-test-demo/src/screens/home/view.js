import React from 'react'
import { connect } from 'dva'
import { Button, Select } from 'antd'
import { translate } from 'react-i18next'

import { key, AppConst } from '../../configs'
import './style.less'

const { Option } = Select

export class HomeView extends React.Component {
  componentDidMount() {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init'
    })
  }

  /**
   * Handle logout
   */
  logout = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout'
    })
  }

  /**
   * Handle change language
   */
  changeLanguage = (lng) => {
    const { i18n } = this.props
    i18n.changeLanguage(lng)
  }

  render() {
    const { user, t, i18n } = this.props
    const currentLanguage = i18n.language
    if (!user) {
      return null
    }
    return (
      <div className="text-center">
        <header>
          <div className="top-left">
            <Select defaultValue={currentLanguage} onChange={value => this.changeLanguage(value)}>
              {
                AppConst.nation.map(item => (
                  <Option key={item.code} value={item.language}>{item.description}</Option>
                ))
              }
            </Select>
          </div>
        </header>
        <h2 className="title">
          {t(key.welcome)} {user.name}
        </h2>
        <Button type="danger" onClick={this.logout}>{t(key.logout)}</Button>
      </div>
    )
  }
}

export default connect(({ home }) => ({ user: home.user }))(translate(key.translations)(HomeView))
