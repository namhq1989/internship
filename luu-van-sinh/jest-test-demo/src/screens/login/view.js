import React from 'react'
import { connect } from 'dva';
import { Form, Icon, Input, Button, Row, Col } from 'antd'
import { Link } from 'react-router-dom'
import { translate } from 'react-i18next'

import { ImageConst, AppConst, key, URLConst } from '../../configs'
import { RcCountryFlag, RcNotification } from '../../components'
import { validate } from './validator'
import './style.less'
import RcAccountKitLogin from '../../components/account-kit-login';

const FormItem = Form.Item

export class LoginView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  /**
   * Handle change form input
   */
  handleChange = (e) => {
    const newState = {}
    newState[e.target.name] = e.target.value
    this.setState(newState)
  }

  /**
   * Handle submit form
   */
  submitLogin = () => {
    const { email, password } = this.state
    const { dispatch, t } = this.props
    const { isValid, message } = validate(email, password)
    if (isValid) {
      dispatch({
        type: 'login/login',
        payload: {
          email,
          password
        }
      })
    } else {
      RcNotification(t(`${key.messages}:${message}`), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="page">
        <Row type="flex" justify="center">
          <Col md={8} xs={18} sm={18} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="" />
            </div>
            <Form className="form-container">
              <FormItem>
                <Input
                  prefix={<Icon type="user" />}
                  type="email"
                  placeholder={t(key.email)}
                  name="email"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t(key.password)}
                  name="password"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem className="btn-center">
                <Button
                  type="primary"
                  htmlType="button"
                  className="btn-submit"
                  onClick={this.submitLogin}
                >
                  {t(key.login)}
                </Button>
                <Row>
                  <RcAccountKitLogin t={t} />
                </Row>
                <Row gutter={16}>
                  {t(key.dontHaveAccount)}
                  <Link to={URLConst.register}>{t(key.signUp)}</Link>
                </Row>
              </FormItem>
            </Form>
          </Col>
        </Row>
        <Row type="flex" justify="center" className="options-language">
          {
            AppConst.nation.map(item => (
              <RcCountryFlag
                changeLanguage={changeLanguage}
                language={item.language}
                code={item.code}
                key={item.code}
              />
            ))
          }
        </Row>
      </div>
    )
  }
}
export default connect()(translate([key.translations, key.messages])(LoginView))
