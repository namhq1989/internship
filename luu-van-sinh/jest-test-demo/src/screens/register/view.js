import React from 'react'
import { Form, Icon, Input, Button, Row, Col } from 'antd'
import { connect } from 'dva'
import { translate } from 'react-i18next'
import { Link } from 'react-router-dom'
import { ImageConst, AppConst, URLConst, key } from '../../configs'
import { validate } from './validator'
import { RcCountryFlag, RcNotification } from '../../components'
import './style.less'

const FormItem = Form.Item

export class RegisterView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      age: '',
      email: '',
      password: '',
    }
  }

  /**
   * Handle change form input
   */
  handleChange = (e) => {
    const newState = {}
    newState[e.target.name] = e.target.value
    this.setState(newState)
  }

  /**
   * Handle submit register
   */
  submitRegister = () => {
    const { email, password, age, name } = this.state
    const { dispatch, t } = this.props
    const { isValid, message } = validate(name, age, email, password)
    if (isValid) {
      dispatch(
        {
          type: 'register/register',
          payload: { email, password, name, age }
        }
      )
    } else {
      RcNotification(t(`${key.messages}:${message}`), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="page">
        <Row type="flex" justify="center">
          <Col md={8} xs={18} sm={18} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="" />
            </div>
            <Form className="form-container">
              <FormItem>
                <Input
                  prefix={<Icon type="user" />}
                  type="text"
                  placeholder={t(key.name)}
                  name="name"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="solution" />}
                  type="number"
                  placeholder={t(key.age)}
                  name="age"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="mail" />}
                  type="email"
                  placeholder={t(key.email)}
                  name="email"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t(key.password)}
                  name="password"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="btn-center">
                <Button
                  type="primary"
                  htmlType="button"
                  className="btn-submit"
                  onClick={this.submitRegister}
                >
                  {t(key.signUp)}
                </Button>
                <Row gutter={16}>
                  {t(key.haveAccount)}
                  <Link to={URLConst.login}>{t(key.login)}</Link>
                </Row>
              </FormItem>
            </Form>
          </Col>
        </Row>
        <Row type="flex" justify="center" className="options-language">
          {
            AppConst.nation.map(item => (
              <RcCountryFlag
                changeLanguage={changeLanguage}
                language={item.language}
                code={item.code}
                key={item.code}
              />
            ))
          }
        </Row>
      </div>
    )
  }
}
export default connect()(translate([key.translations, key.messages])(RegisterView))
