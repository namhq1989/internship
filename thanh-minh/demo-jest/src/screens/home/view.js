import React from 'react'
import { connect } from 'dva'
import { Row, Col, Button } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst } from '../../configs'
import './style.less'

class HomePageView extends React.Component {
  componentDidMount() {
    this.init()
    this.saveUserInfo()
  }

  init = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init',
    })
  }

  saveUserInfo = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/userInfoSave',
    })
  }

  handleLogOut = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout',
    })
  }

  render() {
    const { home: { userInfo } } = this.props
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div>
        <Row type="flex" justify="center">
          <Col span={8} className="wrapper">
            <h1>{t('welcome-page')} <strong className="user-info">{userInfo.name}</strong>{t('come-page')}</h1>
            <Button
              type="primary"
              onClick={this.handleLogOut}
              className="logout-button"
            >{t('log-out')}
            </Button><br /><br />
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}

export default connect(({ home }) => ({ home }))(translate('translations')(HomePageView))
