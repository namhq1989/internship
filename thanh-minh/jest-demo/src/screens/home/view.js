import React from 'react'
import { connect } from 'dva'
import { Row, Col, Button } from 'antd'
import { translate } from 'react-i18next'
import { LanguageChange } from '../../components'
import { KeysConst } from '../../configs/locale'
import './style.less'

export class HomePageView extends React.Component {
  componentDidMount() {
    this.init()
    this.saveUserInfo()
  }

  /**
   * Call to model for Check whether the user account is logged on
   */
  init = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init',
    })
  }

  /**
   * Call to model for save user account into store
   */
  saveUserInfo = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/userInfoSave',
    })
  }

  /**
   * Call to model for log out
   */
  handleLogOut = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout',
    })
  }

  /**
   * Render register view
   */
  render() {
    const { t, i18n, userInfo } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div>
        <Row type="flex" justify="center">
          <Col span={8} className="wrapper">
            <h1>{t(KeysConst.welcomePage)}
              <strong className="user-info">
                {userInfo.name}
              </strong>
              {t(KeysConst.comePage)}
            </h1>
            <Button
              type="primary"
              onClick={this.handleLogOut}
              className="logout-button"
            >{t(KeysConst.logOut)}
            </Button><br /><br />
            <LanguageChange
              t={t}
              changeLanguage={changeLanguage}
            />
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect(
  ({ home }) => (
    { userInfo: home.userInfo }
  )
)(translate('translations')(HomePageView))
