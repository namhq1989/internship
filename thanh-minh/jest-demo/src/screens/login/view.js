import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Button, Input } from 'antd'
import { translate } from 'react-i18next'
import AccountKit from 'react-facebook-account-kit'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification } from '../../components'
import { loginValidator } from './validator'
import './style.less'

const FormItem = Form.Item
export class LoginView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (event) => {
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  submitLogin = () => {
    const { t, dispatch } = this.props
    const { email, password } = this.state
    const validator = loginValidator(email, password)
    if (validator.status) {
      const user = { email, password }
      dispatch({
        type: 'login/login',
        payload: user
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-login">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
            </div>
            <Form className="form-login">
              <FormItem>
                <Input
                  prefix={<Icon type="user" />}
                  id="email"
                  type="email"
                  placeholder={t('email-palceholder')}
                  name="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="lock" />}
                  id="password"
                  type="password"
                  placeholder={t('password-placeholder')}
                  name="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem className="div-login-button">
                <Button
                  type="primary"
                  htmlType="button"
                  className="button-login"
                  onClick={this.submitLogin}
                >
                  {t('login-button')}
                </Button>
              </FormItem>
            </Form>
            <div className="account-kit">
              <AccountKit
                appId="195477857985857"
                version="v1.0"
                onResponse={resp => console.log(resp)}
                csrf="Demo-Account-Kit"
                language={t('language-o')}
                loginType="EMAIL"
              >
                {p => <Button {...p}>Account Kit</Button>
              }
              </AccountKit>
            </div>
            <div className="div-login">
              <span>{t('Do not have an account')}&nbsp;&nbsp;</span>
              <a href="/#/register" title="register">
                {t('Sign-up')}
              </a>
            </div>
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button id="vi-button" onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button id="en-button" onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(LoginView))
