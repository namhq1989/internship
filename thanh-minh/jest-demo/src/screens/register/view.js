import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Input, Button } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification } from '../../components'
import { registerValidator } from './validator'
import './style.less'


const FormItem = Form.Item
class RegisterView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      password: '',
      rePassword: '',
      age: ''
    }
  }

  handleChange = (event) => {
    event.preventDefault()
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  submitRegister = () => {
    const { email, password, rePassword, name, age } = this.state
    const { t, dispatch } = this.props
    const validator = registerValidator(name, email, password, rePassword, age)
    if (validator.status) {
      const account = { name, email, password, rePassword, age }
      dispatch({
        type: 'register/register',
        payload: account
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-register">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
              <h2 className="title-register">{t('Sign-up')}</h2>
            </div>
            <Form className="form-register">
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="user" />}
                  type="text"
                  placeholder={t('Username-placeholder')}
                  name="name"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="mail" />}
                  type="email"
                  placeholder={t('email-palceholder')}
                  name="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t('password-placeholder')}
                  name="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t('repassword-placeholder')}
                  name="rePassword"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="idcard" />}
                  type="text"
                  placeholder={t('age-placeholder')}
                  name="age"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="div-register-button">
                <Button
                  type="pi18nKeyrimary"
                  htmlType="button"
                  className="button-register"
                  onClick={this.submitRegister}
                >
                  {t('Sign-up')}
                </Button>
              </FormItem>
            </Form>
            <div className="div-login">
              <span>{t('Do have an account')}&nbsp;&nbsp;</span>
              <a href="/#/login" title="login">
                {t('login-button')}
              </a>
            </div>
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(RegisterView))
