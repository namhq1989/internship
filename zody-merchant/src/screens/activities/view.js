import React from 'react'
import { Layout, Menu, Icon, Row } from 'antd'
import { ImageConst, AppConst } from '../../configs'
import { StatisticalCard } from '../statistic'
import { RcRangePicker, RcCustomerType } from '../../components'
import style from './style.css'

const { Sider, Header, Content } = Layout
class SideBarView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      collapsed: false,
      collapsedMoblie: false
    }
  }

  componentDidMount() {
    this.handleResize()
    window.addEventListener('resize', this.handleResize)
  }

  getScreenSize = () => {
    const screenPx = window.innerWidth
    if (screenPx <= AppConst.screens['xs-max']) return 'xs'
    if ((screenPx >= AppConst.screens['sm-min'])
      && (screenPx <= AppConst.screens['sm-max'])) return 'sm'
    if ((screenPx >= AppConst.screens['md-min'])
      && (screenPx <= AppConst.screens['md-max'])) return 'md'
    if ((screenPx >= AppConst.screens['lg-min'])
      && (screenPx <= AppConst.screens['lg-max'])) return 'lg'
    if (screenPx >= AppConst.screens['xl-min']) return 'xl'
  }

  isMoblieSide = () => {
    return ['sm', 'xs'].includes(this.getScreenSize())
  }

  handleResize = () => {
    this.setState({
      collapsedMoblie: this.isMoblieSide(),
    })
  }

  toggle = () => {
    const { collapsed } = this.state
    this.setState({
      collapsed: !collapsed,
    })
  }


  render() {
    const { collapsed, collapsedMoblie } = this.state
    return (
      <Layout>
        <Sider
          className="app-sidebar"
          trigger={null}
          collapsible
          collapsed={collapsed}
          width={collapsedMoblie ? 70 : 200}
          collapsedWidth={collapsedMoblie ? 0 : 80}
        >
          <img
            className={style.appLogo}
            src={ImageConst.logoZody}
            alt=""
          />
          <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
            <Menu.Item key="1">
              <Icon type="solution" />
              <span>
                Lịch sử tích điểm
              </span>
            </Menu.Item>
            <Menu.Item key="2">
              <Icon type="logout" />
              <span>
                Dang xuat
              </span>
            </Menu.Item>
          </Menu>
        </Sider>
        <Layout>
          <Header style={{ background: '#fff', padding: 0 }}>
            <Icon
              className={style.trigger}
              type={collapsed ? 'menu-unfold' : 'menu-fold'}
              onClick={this.toggle}
            />
          </Header>
          <Content className={style.content}>
            <Row>
              <RcRangePicker />
              <RcCustomerType />
            </Row>
            <Row gutter={16}>
              <div className={style.sectionTitle}>
                <h4>
                  Thống kê
                </h4>
                <StatisticalCard
                  title="Lượt giao dịch"
                  desc="Theo mốc thời gian"
                  icon={ImageConst.imageDolar}
                  number={10000}
                />
                <StatisticalCard
                  title="Doanh thu"
                  desc="Theo mốc thời gian"
                  icon={ImageConst.imageGiaoDich}
                  number={10000}
                />
                <StatisticalCard
                  title="Zcoin tặng thành công"
                  desc="Theo mốc thời gian"
                  icon={ImageConst.imageZcoin}
                  number={10000}
                />
              </div>
            </Row>
          </Content>
        </Layout>
      </Layout>
    )
  }
}
export default SideBarView
