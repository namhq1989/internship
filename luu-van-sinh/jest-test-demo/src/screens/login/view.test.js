import React from 'react'
import { shallow } from 'enzyme'
import { LoginView } from './view'

const mockFn = jest.fn()
const wrapper = shallow(<LoginView t={mockFn} dispatch={mockFn} />)

test('render correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should respond to change event and change the state of the view', () => {
  const target = { name: 'email', value: 'example@gmail.com' }
  wrapper.find('Input[name="email"]').simulate('change', { target })
  expect(wrapper.state('email')).toEqual('example@gmail.com')
})

test('call handleSubmit function', () => {
  wrapper.find('Button.btn-submit').simulate('click')
})

test('dispatch in handleSubmit function when data is valid', () => {
  const emailInput = { name: 'email', value: 'example@gmail.com' }
  const passwordInput = { name: 'password', value: '111111' }
  wrapper.find('Input[name="email"]').simulate('change', { target: emailInput })
  wrapper.find('Input[name="password"]').simulate('change', { target: passwordInput })
  wrapper.find('Button.btn-submit').simulate('click')
})
