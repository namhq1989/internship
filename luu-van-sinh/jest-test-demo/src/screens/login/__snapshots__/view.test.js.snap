// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="page"
>
  <Row
    gutter={0}
    justify="center"
    type="flex"
  >
    <Col
      className="container"
      md={8}
      sm={18}
      xs={18}
    >
      <div
        className="logo"
      >
        <img
          alt=""
          src={null}
        />
      </div>
      <Form
        className="form-container"
        hideRequiredMark={false}
        layout="horizontal"
        onSubmit={[Function]}
        prefixCls="ant-form"
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            disabled={false}
            name="email"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="user"
              />
            }
            prefixCls="ant-input"
            type="email"
          />
        </FormItem>
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            disabled={false}
            name="password"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="lock"
              />
            }
            prefixCls="ant-input"
            type="password"
          />
        </FormItem>
        <FormItem
          className="btn-center"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Button
            className="btn-submit"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
            type="primary"
          />
          <Row
            gutter={0}
          >
            <RcAccountKitLogin
              t={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "email",
                    ],
                    Array [
                      "password",
                    ],
                    Array [
                      "login",
                    ],
                    Array [
                      "dontHaveAccount",
                    ],
                    Array [
                      "signUp",
                    ],
                  ],
                }
              }
            />
          </Row>
          <Row
            gutter={16}
          >
            <Link
              replace={false}
              to="/register"
            />
          </Row>
        </FormItem>
      </Form>
    </Col>
  </Row>
  <Row
    className="options-language"
    gutter={0}
    justify="center"
    type="flex"
  >
    <RcCountryFlag
      changeLanguage={[Function]}
      code="VN"
      key="VN"
      language="vi"
    />
    <RcCountryFlag
      changeLanguage={[Function]}
      code="US"
      key="US"
      language="en"
    />
  </Row>
</div>
`;
